mlflow/pyfunc/__init__.py

```diff
@@ -1047,10 +1058,17 @@ def spark_udf(spark, model_uri, result_type="double", env_manager="local"):
 
     if not should_use_spark_to_broadcast_file:
         # Prepare restored environment in driver side if possible.
+        # Note: In databricks runtime, because databricks notebook cell output cannot capture
+        # child process output, so that set capture_output to be True so that when conda prepare
+        # env command failed, the exception message will include command stdout/stderr output.
+        # otherwise user have to check cluster driver log to find command stdout/stderr output.
         if env_manager is _EnvManager.CONDA:
             _get_flavor_backend(
-                local_model_path, env_manager=_EnvManager.CONDA, install_mlflow=False
-            ).prepare_env(model_uri=local_model_path, capture_output=False)
+                local_model_path,
+                env_manager=_EnvManager.CONDA,
+                install_mlflow=False,
+                env_root_dir=conda_env_root_dir,
+            ).prepare_env(model_uri=local_model_path, capture_output=is_in_databricks_runtime())
```

Comment:
Is `capture_output=True` useful for non-Databricks use cases?

Reply:
See my explanation here.
https://github.com/mlflow/mlflow/pull/5561#discussion_r842723083

Reply:
Can we update the inline comment to explain why we prefer `False` outside of Databricks?